bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -113,10 +139,17 @@ private State getState(String type, String value) {
 
     private String getItemName(String topic) {
         String[] topicElements = getTopicElements(topic);
-        if (topicElements.length != 4)
+        if (topicElements.length < 4)
             throw new IllegalArgumentException("Event creation failed, invalid topic: " + topic);
         return topicElements[2];
     }
+    
+    private String getMemberName(String topic) {
+        String[] topicElements = getTopicElements(topic);
+        if (topicElements.length != 5)
```

Comment:
The old implementation checked if the length equals 4. Now, that the topic has one more part, I check if the length equals 5. I thought if the topic has more than 5 parts something is wrong with the topic and a Exception should be thrown because there are no topics which have more than 5 parts